data/data/gcp/variables-gcp.tf

```
@@ -3,6 +3,12 @@ variable "gcp_project_id" {
   description = "The target GCP project for the cluster."
 }
 
+variable "gcp_network_project_id" {
+  type        = string
+  description = "The project that the network and subnets exist in when they are not in the main ProjectID."
+  default     = null
```

You check for the empty string `""` when deciding whether to create a resource or not. Shouldn't this be 
```suggestion
  default     = ""
```

---

O believe that the value coming from install config would be an empty string. That may be the one that I need to change if any. The default here follows the convention used in the tf file.